### PR TITLE
fix clientside prediction issue that caused client to see non-existant projectiles

### PR DESCRIPTION
**Pull request by AutumnNova**
also fixed a thing that allowed you to overcharge batteries using the plasma refill script

### CODE REVIEW

**Comment by Aranclanos** [comment]
made a build and tested this, a couple things that need to be fixed:
-pkas dont fire at all
-energy guns dont swap firemode
-energy guns dont fire at all on clients

**Comment by AutumnNova** [comment]
curious

**Comment by corp-0** [comment]
I'm adding a DNM so ir doesn't get accidentally merged without testing. 

Anyone with the time to test if the issues described by Aran are solved now?

**Comment by NoooneyDude** [comment]
Are the issues present in `develop` already? I merged #5546 without testing so it may need to be reverted.

**Comment by NoooneyDude** [comment]
I'll give testing them a go once I get my PR sorted.

**Comment by AutumnNova** [comment]
all of those issues have been resolved by my lastest commit, do note that my branch doesnt have some values in it, so the secondary fire mightnt work on energyguns, develop does have these values so its pointless to add them in a commit here, but adding them in will fix that

also the issues mentioned by aran were reproducable in the editor before i fixed them, most of them were a result of me forgetting to override methods

**Comment by AutumnNova** [comment]
this should now fix the issues 5546 had

**Comment by NoooneyDude** [comment]
Tested on headless with second client. No noticeable issues.

File: UnityProject/Assets/Scripts/Items/Weapons/GunTrigger.cs [hunk 1]
```diff
@@ -33,18 +34,18 @@ public void Awake()
...
-    		   	gun.ServerShoot(shotBy , target, damageZone, true);
-                Chat.AddActionMsgToChat(
-	            shotBy,
-			    "You fumble up and shoot yourself!",
-			    $"{shotBy.ExpensiveName()} fumbles up and shoots themself!");
+    		   	    gun.ServerShoot(shotBy , target, damageZone, true);
+                    Chat.AddActionMsgToChat(
+	                shotBy,
+			        "You fumble up and shoot yourself!",
+			        $"{shotBy.ExpensiveName()} fumbles up and shoots themself!");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Weapons/GunTrigger.cs:41]: The indentation in this area is pretty wack. Is spaces and not tabs. And other areas too, it seems.

**Comment by corp-0** [comment]
@NoooneyDude I see you removed the DNM label. What's the state of this one? Is it good to merge after the indentation is done?

**Comment by NoooneyDude** [comment]
Yep, should be good to merge.

**Pull request merged**